setup.py

```diff
@@ -18,7 +18,7 @@ def get_version():
         exec(version_file.read(), _globals)
     return _globals["__version__"]
 
-requirements = ['pyyaml', 'six', 'uproot']
+requirements = ['pyyaml', 'six', 'uproot>=4.0.7', 'uproot3']
```

Comment:
I'm curious what is the advantage of including both of these?  I think we should just switch to uproot 4 as the only uproot and bump the major version number of curator instead?

Reply:
That is currently the only way to get `numentries`.
uproot4 has currently no such functionality (see [issue  197](https://github.com/scikit-hep/uproot4/issues/197)), so uproot4 + uproot3 is needed at the moment.

Reply:
But bumping the major version is certainly needed as well - first you will need to make sure that all current depends restrict the fast_carpenter version to below that.

Reply:
Ah, I totally missed this with uproot 4! I'm wondering on the trade-off though between requiring both uproot 3 and 4 just for this, or accepting this will be slower and doing:
```
file = uproot.open(filename)
tree = file[treename]
nentries = tree.num_entries
```
What do you think?

Reply:
It depends on how it scales. Since we have pytests I can run them with your suggestion and with uproo3 to compare.

Reply:
From first test: uproot3.numentries provides much more utility (e.g. error handling) than I can simply reproduce with
```python
def _numentries(files, tree, total=False):
    if total:
        return sum(uproot.open(f)[tree].num_entries for f in files)
    results = {}
    for f in files:
        results[f] = uproot.open(f)[tree].num_entries
```

```
E           uproot.exceptions.KeyInFileError: not found: 'events' (with any cycle number)
E
E               Available keys: (none!)
E
E           in file /home/kreczko/workspace/fast-curator/tests/dummy_files/no-tree.root
```

All the different cases and optimizations would need to be covered &rarr; better to use uproot3 until an uproot4 version exists or take some time to re-implement things properly.

Reply:
Ok, makes sense - thanks for testing it. I think this is fine to merge in then.